ketohub/spiders/raw_content_spider.py

```diff
@@ -29,30 +31,46 @@ def _ensure_directory_exists(directory_path):
     os.makedirs(directory_path)
 
 
-def _write_to_file(filepath, content):
-    """Writes the response content to a local file."""
+def _write_to_file(filepath, filename, content):
+    """Writes content to a local file."""
     _ensure_directory_exists(os.path.dirname(filepath))
-    open(os.path.join(filepath, 'index.html'), 'w').write(content)
+    open(os.path.join(filepath, filename), 'w').write(content)
 
 
 class RawContentSpider(spiders.CrawlSpider):
```

Comment:
I think we're running into awkwardness with our tests because we're in the unusual situation of having a class where we don't control instantiation, so we can *only* use constructors with no parameters.

This forced us into a weird situation with the base class, because `download_recipe_contents` doesn't *really* have any reason to be public except that we don't want to try to test it through public methods in each of its subclasses.

It didn't occur to me earlier, but I think the solution is to go the route of containment rather than inheritance. I think we'd solve a lot of the testing awkwardness if, instead of `RawContentSpider`, it was like `PageContentDownloader`. Instead of the spiders inheriting from it, they'd just each hold an instance to one. They'd define their own callbacks, but they'd mostly just be delegating to `PageContentDownloader`, like passing it page and image data for it to persist to disk.

That way, we don't have to struggle to test the downloading functionality because we can just test it directly and then we test each of the spiders using a mock version of `PageContentDownloader`.